internal/go_repository_cache.bzl

```diff
@@ -59,6 +59,12 @@ def _go_repository_cache_impl(ctx):
     cache_env = {
         "GOROOT": go_root,
         "GOCACHE": go_cache,
+
+        # Since v1.21.0, set GOTOOLCHAIN to "local" to use the current toolchain
+        # and avoid downloading other toolchains.
+        #
+        # See https://go.dev/doc/toolchain for more info.
+        "GOTOOLCHAIN": "local",
```

Comment:
Maybe we need to keep the value of `GOTOOLCHAIN` when we use the `host` version in `go_sdk`.

Reply:
Could you add a TODO to investigate this? `go_sdk.host` is so niche that I wouldn't worry about this here.